make.py

```diff
@@ -78,52 +92,125 @@ def make_github_issue_no_notify(title, body, labels):
+        print(f'Could not create Issue "{title}"')
         print("Response:", response.content)
 
 
-template = Path("./issues/test_uncaught.markdown").read_text()
-
-for domain in [
-    # "abode",
-    # "cast",
-    # "config",
-    "deconz",
-    "default_config",
-    "demo",
-    "discovery",
-    "dsmr",
-    "dynalite",
-    "dyson",
-    "gdacs",
-    "geonetnz_quakes",
-    "homematicip_cloud",
-    "hue",
-    "ios",
-    "local_file",
-    "meteo_france",
-    "mikrotik",
-    "mqtt",
-    "plex",
-    "qwikswitch",
-    "rflink",
-    "samsungtv",
-    "tplink",
-    "tradfri",
-    "unifi_direct",
-    "upnp",
-    "vera",
-    "wunderground",
-    "yr",
-    "zha",
-    "zwave",
-]:
-    title = f"Fix {domain} tests that have uncaught exceptions"
-    body = template.replace("{{ DOMAIN }}", domain)
-    labels = [f"integration: {domain}", "to do"]
-    # print(title)
-    # print(body)
-    # print(labels)
-    make_github_issue(title=title, body=body, labels=labels)
+def common_issue_options(func):
+    """Supply common issue options."""
+    func = click.option(
+        "-t",
+        "--token",
+        prompt=True,
+        hide_input=True,
+        help="Set the auth token.",
+    )(func)
+    func = click.option(
+        "-R",
+        "--repo",
+        default=REPO_NAME,
+        show_default=True,
+        help="Set the target repo.",
+    )(func)
+    func = click.option(
+        "-u",
+        "--username",
+        required=True,
+        help="Set the username.",
+    )(func)
+    func = click.option(
+        "-O",
+        "--owner",
+        default=REPO_OWNER,
+        show_default=True,
+        help="Set the repository owner.",
+    )(func)
+    func = click.option(
+        "-T",
+        "--title",
+        required=True,
+        help="Set the issue title.",
+    )(func)
+    func = click.option(
+        "-b",
+        "--body",
+        type=click.Path(exists=True, dir_okay=False),
+        help="Set file path to a markdown file with issue body.",
+    )(func)
+    func = click.option(
+        "-a",
+        "--assignee",
+        show_default=True,
+        help="Set the issue assignee.",
+    )(func)
+    func = click.option(
+        "-m",
+        "--milestone",
+        help="Set the issue milestone.",
+    )(func)
+    func = click.option(
+        "-d",
+        "--domains",
+        is_flag=True,
+        help="Create one issue per domain.",
+    )(func)
+    func = click.option("-l", "--labels", multiple=True, help="Set the issue labels.")(
+        func
+    )
+    return func
+
+
+@click.group(
+    options_metavar="", subcommand_metavar="<command>", context_settings=SETTINGS
+)
+def cli():
+    """Batch create github.com issues."""
+
+
+@click.command(options_metavar="<options>")
+@click.option(
+    "-s", "--silent", is_flag=True, help="Make an issue without notifications."
+)
+@common_issue_options
+def issue(silent, owner, repo, token, username, title, body, labels, domains, **kwargs):
```

Comment:
should this be create_issue?

Reply:
Changed the function name but kept the command name to combine better with `make` -> `make issue`.